src/tests/pools/D3M4626TypePool.t.sol

```diff
@@ -0,0 +1,168 @@
+        
+        vm.prank(address(hub)); pool.withdraw(1);
+        
+        assertEq(pool.assetBalance(), 0);
+        assertEq(dai.balanceOf(address(hub)), 1);
+    }
+
+    function test_withdraw_calls_vault_withdraw_vat_caged() public {
+        deal(address(dai), address(pool), 1);
+        vm.prank(address(hub)); pool.deposit(1);
+        
+        vat.cage();
+        vm.prank(address(hub)); pool.withdraw(1);
+
+        assertEq(pool.assetBalance(), 0);
+        assertEq(dai.balanceOf(address(hub)), 1);
+    }
+
+    function test_redeemable_returns_vault() public {
+        assertEq(pool.redeemable(), address(vault));
+    }
+
+    function test_exit_vault_tokens() public {
+        deal(address(dai), address(this), 1e18);
+        vault.deposit(1e18, address(this));
+        uint256 tokens = vault.totalSupply();
+        vault.transfer(address(pool), tokens);
+        assertEq(vault.balanceOf(address(this)), 0);
+        assertEq(vault.balanceOf(address(pool)), tokens);
+
+        end.setArt(tokens);
+        vm.prank(address(hub)); pool.exit(address(this), tokens);
+
+        assertEq(vault.balanceOf(address(this)), tokens);
+        assertEq(vault.balanceOf(address(pool)), 0);
+    }
+
+    function test_quit_moves_balance() public {
+        deal(address(dai), address(this), 1e18);
+        vault.deposit(1e18, address(this));
+        vault.transfer(address(pool), vault.balanceOf(address(this)));
+        assertEq(vault.balanceOf(address(this)), 0);
+        assertEq(vault.balanceOf(address(pool)), 1e18);
+
+        pool.quit(address(this));
+
+        assertEq(vault.balanceOf(address(this)), 1e18);
+        assertEq(vault.balanceOf(address(pool)), 0);
+    }
+
+    function test_assetBalance_gets_vault_balanceOf_pool() public {
+        deal(address(dai), address(this), 1e18);
+        vault.deposit(1e18, address(this));
+        assertEq(pool.assetBalance(), 0);
+        assertEq(vault.balanceOf(address(pool)), 0);
+
+        vault.transfer(address(pool), 1e18);
+
+        assertEq(pool.assetBalance(), 1e18);
+        assertEq(vault.balanceOf(address(pool)), 1e18);
+    }
+
+    function test_maxWithdraw_gets_available_assets_assetBal() public {
+        deal(address(dai), address(this), 1e18);
+        dai.transfer(address(vault), 1e18);
+        assertEq(dai.balanceOf(address(vault)), 1e18);
+        assertEq(vault.balanceOf(address(pool)), 0);
+
+        assertEq(pool.maxWithdraw(), 0);
+    }
+
+    function test_maxWithdraw_gets_available_assets_daiBal() public {
+        deal(address(dai), address(this), 1e18);
+        vault.deposit(1e18, address(this));
+        vault.transfer(address(pool), 1e18);
+        assertEq(dai.balanceOf(address(vault)), 1e18);
+        assertEq(vault.balanceOf(address(pool)), 1e18);
+
+        assertEq(pool.maxWithdraw(), 1e18);
+    }
+
+    function test_maxDeposit_returns_max_uint() public {
+        assertEq(pool.maxDeposit(), type(uint256).max);
+    }
```

Comment:
I think we shouldn't be checking it returns max uint256 as that is a particularity of this vault deployment. We should be checking instead that it returns the same than the vault maxDeposit.
At least the test name shouldn't be tight to the max uint value.
Maybe we can just change the test name and add an extra check that the vault.maxDeposit is also max uint.

Reply:
Why are these being resolved when the requested action is not complete? We let the issue author resolve the issues, so they can confirm the code author does the changes they want.

Reply:
Also think this is resolved. @sunbreak1211